pkg/gui/controllers/helpers/search_helper.go

```diff
@@ -227,6 +227,18 @@ func (self *SearchHelper) OnPromptContentChanged(searchString string) {
 	}
 }
 
+func (self *SearchHelper) ReApplyFilter(context types.Context) {
+	state := self.searchState()
+	if context == state.Context {
+		filterableContext, ok := context.(types.IFilterableContext)
+		if ok {
+			filterableContext.SetSelectedLineIdx(0)
+			_ = filterableContext.GetView().SetOriginY(0)
+			filterableContext.ReApplyFilter()
```

Comment:
If we rename the `ReApplyFilter` method on the `IFilterableContext`, it would probably make sense to move all this logic inside it. Also, would it make sense, in your opinion, to reset the filter in case of an empty resulting list? I'm not sure, but it's an option.

Reply:
I would not reset the filter because there may intermittently be an empty resulting list (I'm thinking of the weirdness that happens when refreshing the files view for example)

Reply:
The files view is not filterable (yet?), so this wouldn't be an issue right now. But I agree, the behavior could be strange in other cases; say you filter the branches list, type so much that no branch shows up, and then the background fetch comes along and refreshes the branches.